source/textInfos/offsets.py

```diff
@@ -337,26 +321,45 @@ def _getWordOffsets(self,offset):
 			#Inject two alphanumeric characters at the end to fix this 
 			lineText+="xx"
 			if NVDAHelper.localLib.calculateWordOffsets(lineText,len(lineText),offset-lineStart,ctypes.byref(start),ctypes.byref(end)):
-				return start.value+lineStart,min(end.value+lineStart,lineEnd)
+				start = start.value
+				end = end.value
+				if self.encoding != self._UNISCRIBE_ENCODING:
+					# We need to convert the uniscribe based offsets to str offsets.
+					offsetConverter = textUtils.WideStringOffsetConverter(lineText)
+					start, end = offsetConverter.wideToStrOffsets(start, end)
+				return (start, end)
 		#Fall back to the older word offsets detection that only breaks on non alphanumeric
+		if self.encoding == self._UNISCRIBE_ENCODING:

```

Comment:
I'm a bit confused about this, line 317 has `if self.useUniscribe` how does this block relate to that one?